src/main/scala/scalismo/mesh/MeshOperations.scala

```diff
@@ -181,24 +181,24 @@ class TriangleMesh3DOperations(private val mesh: TriangleMesh3D) {
   }
 
   /**
-    * Attempts to reduce the number of vertices of a mesh to the given number of vertices.
-    *
-    * @param targetedNumberOfVertices The targeted number of vertices. Note that it is not guaranteed
-    *                                 that this number is reached exactly
-    * @return The decimated mesh
-    */
-  def decimate(targetedNumberOfVertices : Int) : TriangleMesh[_3D] = {
-      val refVtk = MeshConversion.meshToVtkPolyData(mesh)
-      val decimatePro = new vtk.vtkDecimatePro()
-
-      val reductionRate = 1.0 - (targetedNumberOfVertices / mesh.pointSet.numberOfPoints)
-
-      decimatePro.SetTargetReduction(reductionRate)
-
-      decimatePro.SetInputData(refVtk)
-      decimatePro.Update()
-      val decimatedRefVTK = decimatePro.GetOutput()
-      MeshConversion.vtkPolyDataToTriangleMesh(decimatedRefVTK).get
+   * Attempts to reduce the number of vertices of a mesh to the given number of vertices.
+   *
+   * @param targetedNumberOfVertices The targeted number of vertices. Note that it is not guaranteed
+   *                                 that this number is reached exactly
+   * @return The decimated mesh
+   */
+  def decimate(targetedNumberOfVertices: Int): TriangleMesh[_3D] = {
+    val refVtk = MeshConversion.meshToVtkPolyData(mesh)
+    val decimatePro = new vtk.vtkDecimatePro()
+
+    val reductionRate = 1.0 - (targetedNumberOfVertices / mesh.pointSet.numberOfPoints)
```

Comment:
I just tried this out. I had to change the last value to: mesh.pointSet.numberOfPoints.toDouble. Otherwise that fraction is always 0 and reductionRate is always 1.0

Reply:
@DanaRahbani Thanks for testing and pointing this out. This was really a beginner's mistake 😟  The bug should now be fixed and the corresponding test improved.